charts/chronicle/templates/tests/api-test.yaml

```diff
@@ -8,16 +9,99 @@ metadata:
+      {{- if .Values.auth.required }}
+        - name: wait-for-id-provider
+          {{- include "lib.image" (dict "imageRoot" .Values.test.api.image "global" .Values.global ) | nindent 10 }}
+          command: [ "sh", "-ec" ]
+          args:
+            - |
+              echo "Waiting for id-provider to be ready ..."
+              wait-for-it $HOST:$PORT --timeout=0
+              echo "Id-provider is ready. Exiting."
           env:
             - name: HOST
-              value: {{ include "chronicle.service.name" . }}-chronicle-api
+              value: {{ include "chronicle.id-provider.service" . }}
             - name: PORT
-              value: {{ .Values.port | quote }}
-      serviceAccountName:  {{ include "lib.serviceAccountName" . }}
-      restartPolicy: Never
+              value: "{{ .Values.test.idProvider.port }}"
+        - name: token-loader
+          image: alpine/k8s:1.24.13
+          command: [ "sh", "-ec" ]
+          args:
+            - |
+              echo "Waiting to ensure id-provider is ready ..."
+              sleep 20
+              echo "Getting token from id-provider ..."
+              kubectl exec {{ include "chronicle.id-provider.service" . }}-0 -c id-provider -- oauth-token > /shared-data/jwks-token
+              echo "Token loaded. Exiting."
+          volumeMounts:
+            - name: shared-data
+              mountPath: /shared-data
+      {{- end }}
+        - name: wait-for-api
+          {{- include "lib.image" (dict "imageRoot" .Values.test.api.image "global" .Values.global ) | nindent 10 }}
+          command: [ "sh", "-ec" ]
+          args:
+            - |
+              echo "Waiting for API to be ready ..."
+              wait-for-it $HOST:$PORT --timeout=0
+              echo "Api is ready. Exiting."
+          env:
+            - name: HOST
+              value: {{ include "chronicle.api.service" . }}
+            - name: PORT
+              value: "{{ .Values.port }}"
+        - name: get-api-ip
+          image: alpine/k8s:1.24.13
+          command: [ "sh", "-ec" ]
+          args:
+            - |
+              echo "Getting IP address for API ..."
+              kubectl get service {{ include "chronicle.api.service" . }} -o jsonpath='{.spec.clusterIP}' > /shared-data/api-ip
+              echo "Got address. Exiting."
+          volumeMounts:
+            - name: shared-data
+              mountPath: /shared-data
+      containers:
+        - name: test
+          {{- include "lib.image" (dict "imageRoot" .Values.test.api.image "global" .Values.global ) | nindent 10 }}
+          command: [ "sh", "-ec" ]
+          args:
+            - |
+              {{- if .Values.auth.required }}
+              if [ -f "/shared-data/jwks-token" ]; then
+                echo "Found token."
+                sleep 5
+                export TOKEN=$(cat "/shared-data/jwks-token")
+              fi
+              {{- end }}
+              export HOST=$(cat /shared-data/api-ip)
```

Comment:
does this truly require an IP address, and not a hostname?

Reply:
It does, I'm afraid. My first attempt at resolution had it turn IPv6 addresses into something URL-unparseable and, not wanting to further delay release in faffing further with that, I figured I'd stop there and improve it afterward. (As well as, make it handle mutations too so we can drop the node dependency.)

Reply:
https://github.com/btpworks/chronicle/pull/325 allows it to resolve hostnames.

Reply:
Just tested the Chart with this.